common/scala/src/main/scala/whisk/core/entity/WhiskAction.scala

```diff
@@ -163,25 +163,30 @@ case class WhiskAction(
  *
  * exec is typed to CodeExec to guarantee executability by an Invoker.
  *
- * rev is stored as part of the case-class to make action-matching as
- * narrow as possible. version is not enough, because a user might delete
- * an action and recreate it later, effectively resetting the version
- * counter and thus producing "duplicates".
- *
  * @param namespace the namespace for the action
  * @param name the name of the action
  * @param exec the action executable details
+ * @param parameters the set of parameters to bind to the action environment
  * @param limits the limits to impose on the action
  * @param version the semantic version
- * @param rev the revision of the document
+ * @param publish true to share the action or false otherwise
+ * @param annotation the set of annotations to attribute to the action
+ * @throws IllegalArgumentException if any argument is undefined
  */
+@throws[IllegalArgumentException]
 case class ExecutableWhiskAction(
     namespace: EntityPath,
-    name: EntityName,
+    override val name: EntityName,
     exec: CodeExec[_],
+    parameters: Parameters = Parameters(),
     limits: ActionLimits = ActionLimits(),
     version: SemVer = SemVer(),
-    rev: DocRevision = DocRevision()) {
+    publish: Boolean = false,
+    annotations: Parameters = Parameters())
+    extends WhiskActionLike(name) {
```

Comment:
I felt uncomfortable with the difference between `WhiskAction` and `ExecutableWhiskAction` so I opted for a slightly more verbose but satisfying approach.